aten/src/ATen/native/cuda/SegmentReduce.cu

```diff
@@ -0,0 +1,124 @@
+
+#include <ATen/native/SegmentReduce.h>
+
+#include <ATen/ATen.h>
+#include <ATen/NumericUtils.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/detail/KernelUtils.h>
+#include <cub/cub.cuh>
+#include <iostream>
+
+namespace at {
+namespace native {
+
+struct CustomMax {
+  /// Boolean max operator, returns <tt>(a > b) ? a : b</tt>
+  template <typename OutputT>
+  __host__ __device__ __forceinline__ OutputT
+  operator()(const OutputT& a, const OutputT& b) {
+    return at::_isnan(b) ? b : std::max<OutputT>(a, b);
```

Comment:
unfortunately, on ROCm there's a bug that doesn't correctly return first value from std::max, so here you have to check both a and b for _isnan

Reply:
Thanks for the heads up!